napari/layers/utils/stack_utils.py

```diff
@@ -0,0 +1,188 @@
+                i_kwargs[key] = next(val)
+            except StopIteration:
+                raise IndexError(
+                    "Error adding multichannel image with data shape "
+                    f"{data.shape!r}.\nRequested channel_axis "
+                    f"({channel_axis}) had length {n_channels}, but "
+                    f"the '{key}' argument only provided {i} values. "
+                )
+
+        layerdata = (image, i_kwargs, 'image')
+        layerdata_list.append(layerdata)
+
+    return layerdata_list
+
+
+def stack_to_images(stack: Image, axis: int, **kwargs: Dict,) -> List[Image]:
+    """Function to split the active layer into separate layers along an axis
+
+    Parameters
+    ----------
+    stack : napari.layers.Image
+        The image stack to be split into a list of image layers
+    axis : int
+        The axis to split along.
+
+    Returns
+    -------
+    imagelist: list
+        List of Image objects
+    """
+
+    data, meta, _ = stack.as_layer_data_tuple()
+
+    for key in ("contrast_limits", "colormap", "blending"):
+        meta[key] = None
+
+    name = stack.name
+    num_dim = len(data.shape)
+    # n_channels = data.shape[axis]
+
+    if num_dim < 3:
+        warnings.warn(
+            "The image needs more than 2 dimensions for splitting",
+            UserWarning,
+        )
+        return None
+
+    if axis >= num_dim:
+        warnings.warn(
+            "Can't split along axis {}. The image has {} dimensions".format(
+                axis, num_dim
+            ),
+            UserWarning,
+        )
+        return None
+
+    if 'colormap' in kwargs and kwargs['colormap'] is not None:
+        kwargs['colormap'] = itertools.cycle(kwargs['colormap'])
+
+    if meta['rgb']:
+        if axis == (num_dim - 1) or axis == -1:
+            kwargs['rgb'] = False  # split channels as grayscale
+        else:
+            kwargs['rgb'] = True  # split some other axis, remain rgb
+            meta['scale'].pop(axis)
+            meta['translate'].pop(axis)
+    else:
+        kwargs['rgb'] = False
+        meta['scale'].pop(axis)
+        meta['translate'].pop(axis)
+
+    meta.update(kwargs)
+    imagelist = list()
+    layerdata_list = split_channels(data, axis, **meta)
+    for i, tup in enumerate(layerdata_list):
+        idata, imeta, _ = tup
+        layer_name = f'{name} layer {i}'
+        imeta['name'] = layer_name
+
+        image = Image(idata, **imeta)
+
+        imagelist.append(image)
+
+    return imagelist
+
+
+def images_to_stack(
+    images: List[Union[Image, Labels]], axis: int = 0, **kwargs: Dict,
+) -> Image:
+    """Function to combine selected image layers in one layer
+
+    Parameters
+    ----------
+    images : List
+        List of Image Layers
+    axis : int
+        Index to to insert the new axis
+
+    Returns
+    -------
+    stack : napari.layers.Image
+        Combined image stack
+    """
+
+    data, meta, _ = images[0].as_layer_data_tuple()
```

Comment:
```suggestion
    if len(images) > 0:
        data, meta, _ = images[0].as_layer_data_tuple()
    else:
        return None
```
Curious what people think we should do if the list is empty, I suggest returning `None`, @tlambert03 @jni?

Reply:
I think it should raise an exception.  If we return `None`, then the return type should be changed to `Optional[Layer]`, which reminds you that any consuming function needs to make sure it got a response.